drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -49,23 +51,47 @@ local function find_default_endpoint(device, component)
   return res
 end
 
-local function component_to_endpoint(device, component_id)
-  -- Assumes matter endpoint layout is sequentional starting at 1.
-  local ep_num = component_id:match("switch(%d)")
-  return ep_num and tonumber(ep_num) or find_default_endpoint(device, component_id)
+local function create_endpoint_to_component_map(device)
```

Comment:
I think we may want to also do the profile switching behavior in this function rather than in `do_configure` so that the endpoint map and the profile are in sync. Otherwise we could be fixing the endpoint map of an existing device but the profile would be wrong because the device was already configured.

I also think we probably want to do this stuff only once and it should be skipped if the `ENDPOINT_TO_COMPONENT_MAP` already exists.

Reply:
In d70a3099c1abcd571de9b96b22ad399797ffd37a I moved both the component-to-endpoint mapping and the profile switching into one `initialize_switch` function that will be called on `device_init` if there is no `ENDPOINT_TO_COMPONENT_MAP` on the device